Ambiguous call to QLocale::toString()

Version a8cd9ad316334b6d23755d77066cf63792ae8073

- Kubuntu 20.04
- gcc version 9.3.0 (Ubuntu 9.3.0-17ubuntu1~20.04) 
- Qt 5.12.8
- OpenCascade 7.50
- gmio OFF


```bash
../src/src/base/string_utils.cpp: In static member function ‘static QString Mayo::StringUtils::bytesText(uint64_t, const QLocale&)’:
../src/src/base/string_utils.cpp:97:56: error: call of overloaded ‘toString(uint64_t&)’ is ambiguous
   97 |         return tr("%1%2").arg(locale.toString(sizeBytes), tr("B"));
      |                                                        ^

```

--- a/src/app/recent_files.cpp
+++ b/src/app/recent_files.cpp
@@ -100,7 +100,7 @@ QDataStream& operator<<(QDataStream& stream, const RecentFile& recentFile)
 {
     stream << recentFile.filepath;
     stream << recentFile.thumbnail;
-    stream << recentFile.thumbnailTimestamp;
+    stream << (qint64) recentFile.thumbnailTimestamp;
     return stream;
 }
 
@@ -108,7 +108,7 @@ QDataStream& operator>>(QDataStream& stream, RecentFile& recentFile)
 {
     stream >> recentFile.filepath;
     stream >> recentFile.thumbnail;
-    stream >> recentFile.thumbnailTimestamp;
+    reinterpret_cast<qint64&>(recentFile.thumbnailTimestamp);
     return stream;
 }
 

--- a/src/base/string_utils.cpp
+++ b/src/base/string_utils.cpp
@@ -94,15 +94,15 @@ QString StringUtils::bytesText(uint64_t sizeBytes, const QLocale& locale)
 {
     constexpr int oneMB = 1024 * 1024;
     if (sizeBytes < 1024 )
-        return tr("%1%2").arg(locale.toString(sizeBytes), tr("B"));
+        return tr("%1%2").arg(locale.toString((quint64) sizeBytes), tr("B"));
     else if (sizeBytes < oneMB)
-        return tr("%1%2").arg(locale.toString(sizeBytes / 1024), tr("KB"));
+        return tr("%1%2").arg(locale.toString(((quint64) sizeBytes) / 1024), tr("KB"));
     else if (sizeBytes < (10 * oneMB))
         return tr("%1%2").arg(locale.toString(sizeBytes / double(oneMB), 'f', 2), tr("MB"));
     else if (sizeBytes < (100 * oneMB))
         return tr("%1%2").arg(locale.toString(sizeBytes / double(oneMB), 'f', 1), tr("MB"));
     else
-        return tr("%1%2").arg(locale.toString(sizeBytes / oneMB), tr("MB"));
+        return tr("%1%2").arg(locale.toString((quint64) sizeBytes / oneMB), tr("MB"));
 }
 
 void StringUtils::append(QString* dst, const QString& str, const QLocale& locale)

